src/processors/ipfsFile/mediaUploader.ts

```diff
@@ -0,0 +1,96 @@
+import { urlSource } from '../../clients/ipfs';
+import { Table } from '../../db/db';
+import { IPFSFile, IPFSFileUrl } from '../../types/ipfsFile';
+import { Clients, Processor } from '../../types/processor';
+import { ProcessedTrack } from '../../types/track';
+import { Trigger } from '../../types/trigger';
+import { rollPromises } from '../../utils/rollingPromises';
+
+type TrackUrlJoin = ProcessedTrack & Partial<IPFSFileUrl>;
+type MediaType = 'Artwork' | 'Audio';
+
+const notOnIpfs: (media: MediaType) => Trigger<undefined> =
```

Comment:
Could we make this more generic (especially to support other tables, so that we can use it for the other outstanding artist profile picture PR)

ie, change this from MediaType to

```
RecordUrl = {
  table: Table,
  urlField: string,
  cidField: string
}
```